content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst

```diff
@@ -1,87 +1,106 @@
+
+.. _inventory/shipping/invoice-so:
 
-The price is computed when you **save** the sale order or when you click on
-**Set price**.
+Invoice cost on sales order
+===========================
 
-To invoice the price of the delivery charge on the sale order, click on
-**Set price**, it will add a line with the name of the delivery method as
-a product. It may vary from the real price.
+To invoice shipping costs on the sales order, before the item is delivered, go to the
+:menuselection:`Sales app` and select the desired sales order.
 
-When you create the invoice, it will take the price set on the sale
-order.
+On the sales order, click the :guilabel:`Add Shipping` button at the bottom-right corner.
 
-.. image:: invoicing/invoicing01.png
+.. image:: invoicing/add-shipping.png
    :align: center
+   :alt: Click "Add Shipping" button at the bottom right, near the total.
 
-Invoice the real shipping costs
--------------------------------
+In the :guilabel:`Add a shipping method` pop-up window, choose the intended carrier in the
+:guilabel:`Shipping Method` field.
 
-On your sale order, choose the carrier that will be used. Click on
-**Delivery Method** to choose the right one.
+Then, click the :guilabel:`Get Rate` button to the calculate shipping price based on real-time
+shipping data Odoo's shipping carrier integration.
 
-.. image:: invoicing/invoicing05.png
+The :guilabel:`Cost` is automatically calculated using the weight of the items in the order.
+Finally, click the :guilabel:`Add` button to close the window.
+
+.. image:: invoicing/add-a-shipping-method.png
    :align: center
+   :alt: Calculate shipping by selecting a shipping method.
 
-The price is computed when you **save** the sale order. Confirm the sale
-order and proceed to deliver the product.
+.. _inventory/shipping/sales-order:
 
-The real shipping cost is computed when the delivery order is
-validated, you can see the real cost in the chatter of the delivery order.
+On the sales order, the delivery product appears in the :guilabel:`Order Lines` tab, with the
+:guilabel:`Unit Price` set as the shipping cost calculated in the :guilabel:`Add a shipping method`
+pop-up window.
 
-.. image:: invoicing/invoicing02.png
+.. image:: invoicing/delivery-product.png
    :align: center
+   :alt: Show delivery product on the sales order line.
 
-Go back to the sale order, the real cost is now added to the sale
-order.
+Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
+that includes the shipping cost that was added earlier.
 
-.. image:: invoicing/invoicing03.png
-   :align: center
+Invoice real shipping costs
+===========================
 
-When you create the invoice, it will take the price computed by the
-carrier.
+To modify the invoice to reflect the real cost of shipping, follow the steps :ref:`above
+<inventory/shipping/invoice-so>` to create an invoice with a delivery product with a :guilabel:`Unit
+Price` of zero.
```

Comment:
The steps above are about creating an SO, not an invoice, which seems to be what we'd want to do here first.
```suggestion
<inventory/shipping/invoice-so>` to create and and confirm a sales order with a delivery product with a :guilabel:`Unit
Price` of zero.
```

Reply:
I'm not sure if I agree? The steps above take a longer time adding the delivery product on the SO, yes, but the last instruction is to create an invoice. In line 88:
> Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
that includes the shipping cost that was added earlier.